cdrs-tokio/src/cluster/cluster_metadata_manager.rs

```diff
@@ -6,28 +7,115 @@ use tokio::sync::broadcast::Receiver;
 use crate::cluster::{ClusterMetadata, ConnectionManager, Node};
 
 use crate::events::ServerEvent;
+
+use crate::load_balancing::LoadBalancingStrategy;
 use crate::transport::CdrsTransport;
 
+use super::topology::TopologyReader;
+
 // TODO: handle topology changes
-pub struct ClusterMetadataManager<T: CdrsTransport, CM: ConnectionManager<T>> {
-    #[allow(dead_code)]
+pub struct ClusterMetadataManager<
+    T: CdrsTransport + 'static,
+    CM: ConnectionManager<T> + Send + Sync,
+    LB: LoadBalancingStrategy<T, CM> + Send + Sync,
+> {
     event_receiver: Receiver<ServerEvent>,
     metadata: ArcSwap<ClusterMetadata<T, CM>>,
+    topology_reader: TopologyReader<T, CM, LB>,
+    _transport: PhantomData<T>,
 }
 
-impl<T: CdrsTransport, CM: ConnectionManager<T>> ClusterMetadataManager<T, CM> {
+impl<
+        T: CdrsTransport,
+        CM: ConnectionManager<T> + Send + Sync,
+        LB: LoadBalancingStrategy<T, CM> + Send + Sync,
+    > ClusterMetadataManager<T, CM, LB>
+{
     pub fn new(
-        event_receiver: Receiver<ServerEvent>,
         contact_points: Vec<Arc<Node<T, CM>>>,
+        event_receiver: Receiver<ServerEvent>,
+        topology_reader: TopologyReader<T, CM, LB>,
     ) -> Self {
         ClusterMetadataManager {
             event_receiver,
             metadata: ArcSwap::from_pointee(ClusterMetadata::new(contact_points)),
+            topology_reader,
+            _transport: Default::default(),
         }
     }
 
     #[inline]
     pub fn metadata(&self) -> Arc<ClusterMetadata<T, CM>> {
         self.metadata.load().clone()
     }
+
+    async fn perform_refresh(&mut self) {
+        let topo_info = self.topology_reader.read_topology_info().await;
+        println!("{:?}", topo_info);
```

Comment:
It would be good to avoid debugging println from a PR - it might end up in a production version by mistake.